src/multi_box_filter.cpp

```diff
@@ -189,29 +189,37 @@ bool LaserScanMultiBoxFilter::update(const sensor_msgs::LaserScan& input_scan, s
 
 bool LaserScanMultiBoxFilter::inBox(const tf::Point& point)
 {
-  return point.x() < max_.x() && point.x() > min_.x() && point.y() < max_.y() && point.y() > min_.y() &&
-         point.z() < max_.z() && point.z() > min_.z();
+  bool result = false;
+  for (unsigned int i = 0; i < max_.size() && !result; ++i)
+  {
+    result |= point.x() < max_[i].x() && point.x() > min_[i].x() && point.y() < max_[i].y() &&
+              point.y() > min_[i].y() && point.z() < max_[i].z() && point.z() > min_[i].z();
+  }
+  return result;
 }
 
-void LaserScanMultiBoxFilter::updateTfPoints(const Box& box)
+void LaserScanMultiBoxFilter::updateTfPoints(const std::vector<Box>& box_array)
 {
-  max_.setX(box.max.x);
-  max_.setY(box.max.y);
-  max_.setZ(box.max.z);
+  max_.clear();
+  max_.reserve(box_array.size());
+  min_.clear();
+  min_.reserve(box_array.size());
 
-  min_.setX(box.min.x);
-  min_.setY(box.min.y);
-  min_.setZ(box.min.z);
+  for (unsigned int i = 0; i < box_array.size(); ++i)
+  {
+    max_.push_back(tf::Point(box_array[i].max.x, box_array[i].max.y, box_array[i].max.z));
+    min_.push_back(tf::Point(box_array[i].min.x, box_array[i].min.y, box_array[i].min.z));
```

Comment:
Could you try emplace_back instead of push_back?

```suggestion
    max_.emplace_back(box_array[i].max.x, box_array[i].max.y, box_array[i].max.z);
    min_.emplace_back(box_array[i].min.x, box_array[i].min.y, box_array[i].min.z);
```